packages/webgpu/scripts/build/dawn.ts

```diff
@@ -78,53 +80,67 @@ const ios = {
-          ...ios.args,
+          ...apple.args,
         },
         `🍏 ${platform} ${sdk}`,
       );
-      copyLib("ios", platform, sdk);
+      copyLib("apple", platform, sdk);
     }
   }
 
   libs.forEach((lib) => {
     console.log(`Building fat binary for iphone simulator: ${lib}`);
     $(
-      `lipo -create ${projectRoot}/libs/ios/x86_64_iphonesimulator/${lib}.a ${projectRoot}/libs/ios/arm64_iphonesimulator/${lib}.a -output ${projectRoot}/libs/ios/${lib}.a`,
+      `lipo -create ${projectRoot}/libs/apple/x86_64_iphonesimulator/${lib}.a ${projectRoot}/libs/apple/arm64_iphonesimulator/${lib}.a -output ${projectRoot}/libs/apple/${lib}.a`,
     );
   });
 
   libs.forEach((lib) => {
     console.log(`Building fat binary for visionos simulator: ${lib}`);
     $(
-      `lipo -create ${projectRoot}/libs/ios/x86_64_xrsimulator/${lib}.a ${projectRoot}/libs/ios/arm64_xrsimulator/${lib}.a -output ${projectRoot}/libs/ios/${lib}_visionos.a`,
+      `lipo -create ${projectRoot}/libs/apple/x86_64_xrsimulator/${lib}.a ${projectRoot}/libs/apple/arm64_xrsimulator/${lib}.a -output ${projectRoot}/libs/apple/${lib}_visionos.a`,
+    );
+  });
+
+  libs.forEach((lib) => {
+    console.log(`Building fat binary for macOS: ${lib}`);
+    $(
+      `lipo -create ${projectRoot}/libs/apple/x86_64_macosx/${lib}.a ${projectRoot}/libs/apple/arm64_macosx/${lib}.a -output ${projectRoot}/libs/apple/${lib}_macos.a`,
     );
   });
 
   libs.forEach((lib) => {
     console.log(`Building ${lib}`);
     // iOS
-    $(`rm -rf ${projectRoot}/libs/ios/${lib}.xcframework`);
+    $(`rm -rf ${projectRoot}/libs/apple/${lib}.xcframework`);
     $(
       "xcodebuild -create-xcframework " +
-        `-library ${projectRoot}/libs/ios/${lib}.a ` +
-        `-library ${projectRoot}/libs/ios/arm64_iphoneos/${lib}.a ` +
-        ` -output ${projectRoot}/libs/ios/${lib}.xcframework `,
+        `-library ${projectRoot}/libs/apple/${lib}.a ` +
+        `-library ${projectRoot}/libs/apple/arm64_iphoneos/${lib}.a ` +
+        ` -output ${projectRoot}/libs/apple/${lib}.xcframework `,
     );
     // VisionOS
-    $(`rm -rf ${projectRoot}/libs/ios/${lib}_visionos.xcframework`);
+    $(`rm -rf ${projectRoot}/libs/apple/${lib}_visionos.xcframework`);
     $(
       "xcodebuild -create-xcframework " +
-        `-library ${projectRoot}/libs/ios/${lib}_visionos.a ` +
-        `-library ${projectRoot}/libs/ios/arm64_xros/${lib}.a ` +
-        ` -output ${projectRoot}/libs/ios/${lib}_visionos.xcframework `,
+        `-library ${projectRoot}/libs/apple/${lib}_visionos.a ` +
+        `-library ${projectRoot}/libs/apple/arm64_xros/${lib}.a ` +
+        ` -output ${projectRoot}/libs/apple/${lib}_visionos.xcframework `,
+    );
+     // macOS
+     $(
+      "xcodebuild -create-xcframework " +
+        `-library ${projectRoot}/libs/apple/${lib}_macos.a ` +
+        `-library ${projectRoot}/libs/apple/arm64_macosx/${lib}.a ` +
+        ` -output ${projectRoot}/libs/apple/${lib}_macos.xcframework `,
```

Comment:
this probably should be (no fatlib for macos because there is no macos simulator):
```
    // macOS
    $(
      "xcodebuild -create-xcframework " +
        `-library ${projectRoot}/libs/apple/x86_64_macosx/${lib}.a ` +
        `-library ${projectRoot}/libs/apple/arm64_macosx/${lib}.a ` +
        ` -output ${projectRoot}/libs/apple/${lib}_macos.xcframework `,
    );
```